### PR TITLE
fix(security): play framework upgrade

### PR DESCRIPTION
**Pull request by david-leifker**
Update frontend components: pac4j, play
Added gzip filter
Add happy path test for oidc
Added health check at /health (to be consistent with gms)

Ignore the logging changes, those are from [PR](https://github.com/datahub-project/datahub/pull/6616).

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Review by RyanHolstien — APPROVED** [review]

File: datahub-frontend/test/app/ApplicationTest.java [hunk 1]
```diff
@@ -1,11 +1,147 @@
...
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@SetEnvironmentVariable(key = "DATAHUB_SECRET", value = "test")
+@SetEnvironmentVariable(key = "KAFKA_BOOTSTRAP_SERVER", value = "")
+@SetEnvironmentVariable(key = "DATAHUB_ANALYTICS_ENABLED", value = "false")
+@SetEnvironmentVariable(key = "AUTH_OIDC_ENABLED", value = "true")
+@SetEnvironmentVariable(key = "AUTH_OIDC_JIT_PROVISIONING_ENABLED", value = "false")
+@SetEnvironmentVariable(key = "AUTH_OIDC_CLIENT_ID", value = "testclient")
+@SetEnvironmentVariable(key = "AUTH_OIDC_CLIENT_SECRET", value = "testsecret")
+public class ApplicationTest extends WithBrowser {
```
Comment (jjoyce0510) [inline datahub-frontend/test/app/ApplicationTest.java:44]: Thank you so much for adding this! It sets a much better precedent :)

File: datahub-frontend/test/app/ApplicationTest.java [hunk 2]
```diff
@@ -1,11 +1,147 @@
...
+  @BeforeAll
+  public void init() throws IOException, InterruptedException {
+    _gmsServer = new MockWebServer();
+    _gmsServer.enqueue(new MockResponse().setBody("{\"value\":\"urn:li:corpuser:testUser@myCompany.com\"}"));
+    _gmsServer.enqueue(new MockResponse().setBody("{\"accessToken\":\"faketoken_YCpYIrjQH4sD3_rAc3VPPFg4\"}"));
+    _gmsServer.start(gmsServerPort());
+
+    _oauthServer = new MockOAuth2Server();
+    _oauthServer.enqueueCallback(
+            new DefaultOAuth2TokenCallback(ISSUER_ID, "testUser", List.of(), Map.of(
```
Comment (jjoyce0510) [inline datahub-frontend/test/app/ApplicationTest.java:84]: This is so cool.

File: datahub-frontend/test/app/ApplicationTest.java [hunk 3]
```diff
@@ -1,11 +1,147 @@
 package app;
 
-import org.junit.Test;
+import controllers.routes;
+import no.nav.security.mock.oauth2.MockOAuth2Server;
```
Comment (jjoyce0510) [inline datahub-frontend/test/app/ApplicationTest.java:4]: Great find with this Lib!

File: datahub-frontend/conf/application.conf [hunk 4]
```diff
@@ -174,7 +189,8 @@ analytics.enabled = ${?DATAHUB_ANALYTICS_ENABLED}
 
 # Kafka Producer Configuration
 analytics.kafka.bootstrap.server = ${KAFKA_BOOTSTRAP_SERVER}
-analytics.tracking.topic = ${DATAHUB_TRACKING_TOPIC}
+analytics.tracking.topic = DataHubUsageEvent_v1
```
Comment (jjoyce0510) [inline datahub-frontend/conf/application.conf:192]: ty!

File: datahub-frontend/conf/application.conf [hunk 5]
```diff
@@ -19,25 +19,39 @@ play.application.loader = play.inject.guice.GuiceApplicationLoader
...
-play.server.akka.max-header-value-length = ${?DATAHUB_AKKA_MAX_HEADER_VALUE_LENGTH}
+play.server.akka.max-header-size = 8k
+play.server.akka.max-header-size = ${?DATAHUB_AKKA_MAX_HEADER_VALUE_LENGTH}
+
+play.filters {
+  enabled = [
+    play.filters.gzip.GzipFilter
+  ]
+
+  gzip {
```
Comment (jjoyce0510) [inline datahub-frontend/conf/application.conf:44]: Yess! This is going to be a big win!

File: datahub-frontend/app/controllers/AuthenticationController.java [hunk 6]
```diff
@@ -263,23 +277,19 @@ private Result redirectToIdentityProvider() {
         }
 
         try {
-            final HttpAction action = client.redirect(playWebContext);
-            return new PlayHttpActionAdapter().adapt(action.getCode(), playWebContext);
+            final Optional<RedirectionAction> action = client.getRedirectionAction(playWebContext);
+            return new PlayHttpActionAdapter().adapt(action.get(), playWebContext);
```
Comment (jjoyce0510) [inline datahub-frontend/app/controllers/AuthenticationController.java:281]: Minor: If this safe given that we're now dealing with an optional?

File: datahub-frontend/app/auth/sso/oidc/custom/CustomOidcAuthenticator.java [hunk 7]
```diff
@@ -49,14 +49,14 @@ public class CustomOidcAuthenticator implements Authenticator<OidcCredentials> {
...
-  protected OidcClient client;
+  protected OidcClient<OidcConfiguration> client;
 
-  private ClientAuthentication clientAuthentication;
+  private final ClientAuthentication clientAuthentication;
 
-  public CustomOidcAuthenticator(final OidcConfiguration configuration, final OidcClient client) {
-    CommonHelper.assertNotNull("configuration", configuration);
+  public CustomOidcAuthenticator(final OidcClient<OidcConfiguration> client) {
+    CommonHelper.assertNotNull("configuration", client.getConfiguration());
```
Comment (jjoyce0510) [inline datahub-frontend/app/auth/sso/oidc/custom/CustomOidcAuthenticator.java:57]: Nice simplification!

File: datahub-frontend/app/auth/sso/oidc/OidcCallbackLogic.java [hunk 8]
```diff
@@ -149,8 +152,8 @@ private Result handleOidcCallback(final OidcConfigs oidcConfigs, final Result re
 
       // Successfully logged in - Generate GMS login token
       final String accessToken = _authClient.generateSessionTokenForUser(corpUserUrn.getId());
-      context.getJavaSession().put(ACCESS_TOKEN, accessToken);
-      context.getJavaSession().put(ACTOR, corpUserUrn.toString());
+      context.getNativeSession().adding(ACCESS_TOKEN, accessToken);
```
Comment (jjoyce0510) [inline datahub-frontend/app/auth/sso/oidc/OidcCallbackLogic.java:155]: fancy!

**Comment by jjoyce0510** [comment]
One thing I'm seeing is that the Invite Users Smoke test is failing. We should locally verify that the invite users functionality is working before merging this

**Pull request merged**